docs/troubleshoot/app-framework/zss-error-codes.md

```diff
@@ -0,0 +1,371 @@
+
+### ZWES1034E
+
+  Server startup problem: Address '%s' not valid.
+
+  **Reason:**
+
+  IP address is not valid.
+
+  **Action:**
+
+  Use valid IP address, e.g. `0.0.0.0`.
+
+### ZWES1036E
+
+  Server startup problem: Ret='%d', res='0x%x'
+
+  **Reason:**
+
+  Server has failed to start.
+
+  **Action:**
+
+  If the next message is `ZWES1037E` then refer [ZWES1037E](###ZWES1037E).
+
+### ZWES1037E
+
+  This is usually because the server port '%d' is occupied. Is ZSS running twice?
+
+  **Reason:**
+
+  ZSS port number is already occupped. 
+
+  **Action:**
+
+  Check if another ZSS instance is already running, or chose another free port number.
+
+### ZWES1065E
+
+  Failed to configure https server, check agent https settings
+
+  **Reason:**
+
+  Failed to configure https server. 
+
+  **Action:**
+
+  Check agent https settings.
+
+
+## ZSS warning messages
+
+### ZWES1000W
+
+  Privileged server name not provided, falling back to default.
+
+  **Reason:**
+
+  Privileged server name not configured in `zowe.yaml`.
+
+  **Action:**
+
+  If your privileged server name is `ZWESIS_STD` then no action required. Otherwise set `components.zss.crossMemoryServerName` property in `zowe.yaml` to the correct name.
+
+### ZWES1005W
+
+  Plugin ID was not found in '%s'
+
+  **Reason:**
+
+  `pluginId` property wasn't found in `<path-to-pluginDefinition.json>` file.
+
+  **Action:**
+
+  Add `pluginId` property into `<path-to-pluginDefinition.json>` file.
+
+### ZWES1012W
+
+  Could not open pluginsDir '%s': Ret='%d', res='0x%x'
+
+  **Reason:**
+
+  Could not open `<pluginsDir>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  Check that `<pluginsDir>` exists and allows reading.
+
+### ZWES1060W
+
+  Failed to init TLS environment, rc=%d(%s)
+
+  **Reason:**
+
+  Failed to initialized TLS environment.
+
+  **Action:**
+
+  Ensure that ZSS certificate is configured correctly.
+
+### ZWES1103W
+
+  Product Registration failed, RC = %d
+
+  **Reason:**
+
+  Failed to register ZSS.
+
+  **Action:**
+
+  No action required.
+
+### ZWES1201W
+
+  Could not %s file '%s': Ret='%d', res='%d'
+
+  **Reason:**
+
+  Unixfile REST Service could not `<action>` file `<filename>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  No action required.
+
+### ZWES1103W
+
+  Could not get metadata for file '%s': Ret='%d', res='%d'
+
+  **Reason:**
+
+  Unixfile REST Service could not get metadata for file `<filename>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  No action required.
+
+### ZWES1400W
+
+  Non standard class provided for '%s' '%s', ending request...
+
+  **Reason:**
+
+  Non standard class provided in the request for Security service. The request was cancelled.
+
+  **Action:**
+
+  No action required.
+
+
+### ZWES1602W
+
+  JWK is in unrecognized format
+
+  **Reason:**
+
+  JSON Web Key(JWK) is in unrecognized format. 
+
+  **Action:**
+
+  No action required.
```

Comment:
If developer, correct the code to the right format (which is?)
If user, contact whoever wrote the code that sent the JWK since its broken